packages/keystone/lib/Keystone/index.js

```diff
@@ -296,7 +296,7 @@ module.exports = class Keystone {
       throw new Error(`Invalid list name "${key}". List names cannot start with an underscore.`);
     }
 
-    const list = new List(key, compose(config.plugins || [])(config), {
+    const list = new List(key, compose(config.plugins || [])({ ...config, key, keystone: this }), {
```

Comment:
adding `key` and `keystone` allows use of this inside the plugin, not sure if this is a problem.

Reply:
I figured we can best do with changing the compose function, 

we will pass on two args to each plugin, it is upto them what to use. 

```
(config, {listKey, keystone}) => config;
```

caveat is that if we give them `{...config, listKey, keystone } => ?` then they may not return listkey and keystone for next plugin.

Reply:
#3011 achieves this. I have used listKey as this is being used many places. `name`/`listName` may also be more suitable instead of listKey.